h/api/storage.py

```diff
@@ -114,15 +118,44 @@ def create_annotation(request, data):
     # annotation.updated get created.
     request.db.flush()
 
+    update_document_metadata(
+        request.db, annotation, document_meta_dicts, document_uri_dicts)
+
+    return annotation
+
+
+def update_document_metadata(session,
+                             annotation,
+                             document_meta_dicts,
+                             document_uri_dicts):
+    """
```

Comment:
Yeah. On reflection it definitely doesn't make sense as part of the `storage` public API, because a different `storage` implementation other than our Pg one might not need any `update_document_metadata()` at all.

I sort of think it belongs in the same place as `storage.create_annotation() and `update_annotation()` because it's "logic for dealing with the model" in the same way as they do. That argues for a private helper function, I guess.

But I'm fine with putting it in `h.api.models.document` and we do already have a bunch of "logic for dealing with the document models" helper functions in there already